### PR TITLE
update service definitions to support the new way

### PR DESCRIPTION
**Pull request by cordoval**
|Q            |A  |
|---          |---|
|Bug Fix?     |n  |
|New Feature? |n  |
|BC Breaks?   |n  |
|Deprecations?|n  |
|Tests Pass?  |n  |
|Fixed Tickets|   |
|License      |MIT|
|Doc PR       |   |
                   
similar update to https://github.com/symfony/AsseticBundle/pull/343

### CODE REVIEW

File: Resources/config/dbal.xml [hunk 1]
```diff
@@ -43,7 +43,9 @@
             <argument>%doctrine.dbal.connection_factory.types%</argument>
         </service>
 
-        <service id="doctrine.dbal.connection" class="Doctrine\DBAL\Connection" factory-service="doctrine.dbal.connection_factory" factory-method="createConnection" abstract="true" />
+        <service id="doctrine.dbal.connection" class="Doctrine\DBAL\Connection" abstract="true">
```
Comment (Ocramius) [inline Resources/config/dbal.xml]: This change requires locking to `symfony/dependency-injection` at https://github.com/doctrine/DoctrineBundle/blob/c12410ca351c7742ced38fe9441f6b84be467e97/composer.json#L26

**Comment by fabpot** [comment]
@Ocramius I will submit a proper PR later on today... but to make it work, I need to fix an issue in Symfony first.

**Comment by fabpot** [comment]
"Proper" in the sense that it should work for all versions of Symfony (from 2.3 to 2.7).

**Comment by fabpot** [comment]
see #381

**Comment by cordoval** [comment]
closing as we have added the factory conditionally in #381

**Pull request closed without merging**